### PR TITLE
New SLE 12/15 rule all_apparmor_profiles_in_enforce_complain_mode whi…

### PR DESCRIPTION
**Pull request by rumch-se**
…ch covers CIS recommendation

#### Description:

- _This rule covers CIS recommendation 1.7.1.3 Ensure all AppArmor Profiles are in enforce or complain mode_

#### Rationale:

- At the moment CIS profiles for SLE 12/15 does not include this recommendation. The recommendation could be extended to support - other Linux distributions

### CODE REVIEW

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml
Comment (dodys) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml:1]: I believe you can remove: multi_platform_rhel,multi_platform_fedora,multi_platform_ol

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml
Comment (dodys) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml:55]: typo in complian

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml
Comment (dodys) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml:59]: typo in complian

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml
Comment (dodys) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml:35]: typo in complian

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml
Comment (dodys) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml:46]: you can consolidate this is one line:
`<linux:apparmorstatus_object id="apparmor_exists_on_our_system" version="1" />`

**Review by dodys — CHANGES_REQUESTED** [review]
a few fixes needed

**Comment by dodys** [comment]
fyi, since we don't have a SUSE maintainer and this rule also is needed for Ubuntu, I've assigned it to me.

**Comment by dodys** [comment]
This rule is not buildable in Ubuntu (and Debian), as apparmor schema was only added in OVAL Format 5.11.2 and Ubuntu now only has OVAL Format 5.11.1.  A separate OVAL will be sent after this is merged to make it work in Ubuntu.

@rumch-se can you confirm that this is also the case for OpenSUSE, as the logs are showing that it doesn't recognize the apparmor schema.

**Comment by rumch-se** [comment]
Hello @dodys 
Yes I can confirm about openSUSE
"
 AppArmor is installed and running on any installation of openSUSE® Leap by default, regardless of what patterns are installed. The packages listed below are needed for a fully-functional instance of AppArmor:
source: https://doc.opensuse.org/documentation/leap/security/html/book-security/cha-apparmor-start.html 
"

**Comment by rumch-se** [comment]
Hello @dodys 
I have done expected corrections.
According to question related to OVAL version check - in my case the version is 5.11.1 which is < 5.11.2
Have a nice day
Rumen

**Review by dodys — APPROVED** [review]
lgtm, thanks! 
@marcusburghardt this rule will fail to build in a few of the containers because of the OVAL version, as apparmor tests were added to OVAL Format 5.11.2 only. Can we merge it as-is?

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml:10]: Can this and other tasks be executed with the `command` module instead of `shell` module?
The `command` module should be used unless `shell` module is explicitly required.

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml:25]: Here it would be better to use the [assert](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html#ansible-collections-ansible-builtin-assert-module) module.

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml:9]: Could you include the `{{{ rule_title }}}` prefix on tasks names, please?
https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#ansible

File: linux_os/guide/system/apparmor/var_apparmor_mode.var
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/var_apparmor_mode.var:7]: nitpick: I don't have strong opinion on that, but it seems more readable if the break line tag is at the end of the line instead of the beginning of the next line. In any case, I leave this on the authors preference.

File: linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml
Comment (marcusburghardt) [inline linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/oval/shared.xml:68]: Could you include an empty line at the end, please?

**Review by marcusburghardt — CHANGES_REQUESTED** [review]

**Comment by marcusburghardt** [comment]
> lgtm, thanks! @marcusburghardt this rule will fail to build in a few of the containers because of the OVAL version, as apparmor tests were added to OVAL Format 5.11.2 only. Can we merge it as-is?

Is the package already available but the containers are not updated?
If not, is there any plan to update the OVAL version on these systems?

I am not sure about the impact of CI tests if we merge it. @mildas , maybe you can help us on this?

**Comment by rumch-se** [comment]
Good morning @marcusburghardt  and @dodys 
Thank you for your feedback.
The proposed corrections are done.
Have a nice day
Rumen

**Comment by marcusburghardt** [comment]
> > lgtm, thanks! @marcusburghardt this rule will fail to build in a few of the containers because of the OVAL version, as apparmor tests were added to OVAL Format 5.11.2 only. Can we merge it as-is?
> 
> Is the package already available but the containers are not updated? If not, is there any plan to update the OVAL version on these systems?
> 
> I am not sure about the impact of CI tests if we merge it. @mildas , maybe you can help us on this?

I have talked to @mildas offline and he confirmed my concern of merging this PR now. It would impact the CI tests and cause failures in other PRs.

The ideal would be to ensure the OVAL Format 5.11.2 support before merging.

**Comment by marcusburghardt** [comment]
> > > > lgtm, thanks! @marcusburghardt this rule will fail to build in a few of the containers because of the OVAL version, as apparmor tests were added to OVAL Format 5.11.2 only. Can we merge it as-is?
> > > 
> > > 
> > > Is the package already available but the containers are not updated? If not, is there any plan to update the OVAL version on these systems?
> > > I am not sure about the impact of CI tests if we merge it. @mildas , maybe you can help us on this?
> > 
> > 
> > I have talked to @mildas offline and he confirmed my concern of merging this PR now. It would impact the CI tests and cause failures in other PRs.
> > The ideal would be to ensure the OVAL Format 5.11.2 support before merging.
> 
> Should we rename the oval to be sle15.xml so it won't get used in other platforms? Would that fix this issue temporarily?

Interesting idea @dodys . I believe this should work. What do you think @yuumasato ?

**Comment by dodys** [comment]
@rumch-se could you please rename the oval file from `shared.xml` to `sle.xml` ?

**Comment by rumch-se** [comment]
Hello @dodys 
The change was done.
Have a nice day
Rumen

**Comment by dodys** [comment]
> Hello @dodys The change was done. Have a nice day Rumen

Thanks!
We will wait now for the test results to be sure it works. Also do test on your side, but I would expect that an OVAL file named `sle.xml`, should work for you.

**Comment by dodys** [comment]
@marcusburghardt ok, so the renaming of the oval file will not prevent tests to fail.
Then I think the way to go is to drop that oval altogether. What do you think?

**Comment by dodys** [comment]
Hi @rumch-se, could you please rebase this PR and remove the commit to add the OVAL. Since your OVAL is dependent of OVAL format 5.11.2 and most of the CI only has OVAL format 5.11.1, we won't be able to merge it as this will break the CI.
I will later add an SCE audit script for ubuntu.
@teacup-on-rockingchair fyi ^ 
let us know if you have any doubts.

**Comment by rumch-se** [comment]
Hello @dodys 
I have done a rebase and removed the OVAL part.
Have a nice day
Rumen

**Review by dodys — APPROVED** [review]
lgtm thanks!

**Review by marcusburghardt — APPROVED** [review]

**Comment by marcusburghardt** [comment]
The Automatus tests are failing because the rule is restricted to `sle` products.

**Comment by marcusburghardt** [comment]
Although I don't have a system to properly test `apparmor`, this new rule is restricted to `sle` products, was proposed by a known SUSE maintainer and approved by a know Ubuntu maintainer. The files are also sane for me.

Therefore, I am overriding CODEOWNERS since a SUSE approver is not currently available.

**Pull request merged**
